pkg/ebpf/tracer.go

```diff
@@ -264,6 +267,27 @@ func (m *FlowFetcher) Close() error {
 	return errors.New(`errors: "` + strings.Join(errStrings, `", "`) + `"`)
 }
 
+// doIgnoreNoDev runs the provided syscall over the provided device and ignores the error
+// if the cause is a non-existing device (just logs the error as debug).
+// If the agent is deployed as part of the Network Observability pipeline, normally
+// undeploying the FlowCollector could cause the agent to try to remove resources
+// from Pods that have been removed immediately before (e.g. flowlogs-pipeline or the
+// console plugin), so we avoid logging some errors that would unnecessarily raise the
+// user's attention.
+// This function uses generics because the set of provided functions accept different argument
+// types.
+func doIgnoreNoDev[T any](sysCall func(T) error, dev T, log *logrus.Entry) error {
+	if err := sysCall(dev); err != nil {
+		if errors.Is(err, unix.ENODEV) {
+			log.WithError(err).Debug("can't delete. Probably the " +
```

Comment:
Maybe it should be still logged as Error level, but with a message like "blah blah - ignore this error if the container or interface is intentionally being deleted, such as on pod shutdown" or something similar?
My concern is that we may miss legitimate errors.

Reply:
Yeah good appreciation. Changing it.